FilmFestivalLoader/IFFR/parse_iffr_html.py

```diff
@@ -650,42 +667,59 @@ def handle_starttag(self, tag, attrs):
                     self.screened_film_slugs.append(screened_film_slug)
                     self.state_stack.change(self.FilmInfoParseState.IN_SCREENED_FILM_LINK)
             elif tag == 'section':
-                if has_category(self.film, Film.category_combinations) or self.event_is_combi:
-                    self.set_combination()
+                self.finish_film_info()
                 self.state_stack.change(self.FilmInfoParseState.DONE)
 
         # Screening part.
         else:
             self.handle_screening_starttag(tag, attrs, self.state_stack,
-                                           self.FilmInfoParseState.AWAITING_SCREENINGS,
+                                           self.ScreeningParseState.IN_SCREENINGS,
                                            self.FilmInfoParseState.AWAITING_SCREENED_FILM_LINK)
 
     def handle_endtag(self, tag):
         HtmlPageParser.handle_endtag(self, tag)
 
+        # Article part.
         if self.state_stack.state_is(self.FilmInfoParseState.IN_EMPHASIS) and tag == 'em':
             self.state_stack.pop()
         elif self.state_stack.state_is(self.FilmInfoParseState.IN_PARAGRAPH) and tag == 'p':
             self.state_stack.pop()
             self.add_paragraph()
         elif self.state_stack.state_is(self.FilmInfoParseState.IN_ARTICLE) and tag == 'div':
             self.state_stack.pop()
-            self.set_article()
-            self.state_stack.change(self.FilmInfoParseState.AWAITING_SCREENINGS)
+            self.state_stack.change(self.FilmInfoParseState.ARTICLE_DONE)
+
+        # Metadata part.
+        elif self.state_stack.state_is(self.FilmInfoParseState.IN_METADATA) and tag == 'dl':
+            self.state_stack.pop()
+
+        # Combination part.
         elif self.state_stack.state_is(self.FilmInfoParseState.IN_SCREENED_FILM_LIST) and tag == 'a':
             self.state_stack.pop()
         elif self.state_stack.state_is(self.FilmInfoParseState.IN_SCREENED_FILM_LINK) and tag == 'a':
             self.state_stack.change(self.FilmInfoParseState.AWAITING_SCREENED_FILM_LINK)
+
+        # Screening part.
         else:
             self.handle_screening_endtag(tag, self.state_stack, self.FilmInfoParseState.AWAITING_SCREENED_FILM_LINK)
 
     def handle_data(self, data):
         HtmlPageParser.handle_data(self, data)
 
+        # Article part.
         if self.state_stack.state_in([self.FilmInfoParseState.IN_PARAGRAPH,
                                       self.FilmInfoParseState.IN_EMPHASIS,
                                       self.FilmInfoParseState.IN_ARTICLE]):
             self.article_paragraph += data.replace('\n', ' ')
```

Comment:
Please consider to make this a method in `HtmlPageParser` too, so parsing articles is fully abstracted in that class.

Reply:
Done.